torch/utils/checkpoint.py

```diff
@@ -79,21 +104,34 @@ def checkpoint(function, *args):
         grads are needed for model inputs, otherwise the checkpointed part of the
         model won't have gradients.
 
+    .. warning:
+        Checkpointing is implemented by rerunning a forward-pass segment for
+        each checkpointed segment during backward.  This can result in running
+        states like the RNG state used for dropout to be advanced more than
+        they would be without checkpointing, which can cause checkpoints that
+        include dropout invocations to lose end-to-end bitwise accuracy as
```

Comment:
"to lose end-to-end bitwise accuracy as
        compared to non-checkpointed passes"

-> "to have non-deterministic output compared to"